cartography/util.py

```diff
@@ -125,18 +131,21 @@ def aws_handle_regions(func):
         'UnrecognizedClientException',
     ]
 
-    def inner_function(*args, **kwargs):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
         try:
             return func(*args, **kwargs)
         except botocore.exceptions.ClientError as e:
             # The account is not authorized to use this service in this region
             # so we can continue without raising an exception
             if e.response['Error']['Code'] in ERROR_CODES:
                 logger.warning("{} in this region. Skipping...".format(e.response['Error']['Message']))
-                return []
+                return default_return_value
             else:
                 raise
-    return inner_function
+    if func is None:
```

Comment:
Does func specifically need to be `None` or does it simply need to be "falsy"? If the latter, will need `if not func` instead.

Reply:
It makes more sense to be `None` as a the default value for `func`, to say that func was not provided as an argument.
For example this case:
Here we invoke the decorator before actually decorating, and so this first invocation will not have `f` as a supplied argument,
```
@aws_handle_regions(default_return_value=...)
def f():
    pass
```
, rather than this case:
```
@aws_handle_regions
def f():
    pass
```
The only other way for `func` to be falsy is if we specifically made it False. In this case, the decorator won't actually work, because it would try do do `False(*args, **kwargs)`
@aws_handle_regions(False, default_return_value=...)
def f():
    pass